src/lib/JreSetupManager.ts

```diff
@@ -88,17 +103,16 @@ class JreSetupManager extends AsyncCreatable {
         return javaHome;
     }
 
-    private async autoDetectJavaHome(): Promise<string> {
-        return new Promise((resolve) => {
-            findJavaHome({ allowJre: true }, (err: Error, home: string) => {
-                if (err) {
-                    this.logger.trace(`Could not auto-detect Java Home: ${err.message}`);
-                    return resolve(null); // We don't want to reject here. No Java Home is handled later
-                } else {
-                    return resolve(home);
-                }
-            });
-        });
+    private autoDetectJavaHome(): Promise<string> {
+        return this.dependencies.autoDetectJavaHome(this.handleAutoDetectedJavaHome);
+    }
+    handleAutoDetectedJavaHome(err: Error, home: string): string {
+        if (err) {
+            return null; // Absence of javaHome is handled later
```

Comment:
This smells a little funny here. Going to a lot of trouble in defining this function for autoDetectJavaHome just to ignore an error and return either null or the given home param.

Reply:
find-java-home detects Java Home by running parent of parent of javac. If javac does not exist, this call returns an error. To us, this error only means that java has not been setup in the machine. Line 86, 87 already handles this case. Throwing an error here is not going to give us additional information.
I originally had a logger.trace here, but after my recent refactor to unit test this, had to remove it since this is passed as a callback and logger loses its meaning when this function is invoked.

Reply:
I would've ideally liked to invoke `javac` through child_process and found its grandparent directory. Since this find-java-home checks Windows registry as well, this seemed like a good idea. But, in general, find-java-home has been super funky to code or unit test.

https://www.npmjs.com/package/find-java-home